tests/playwright/src/model/ai-lab-page.ts

```diff
@@ -0,0 +1,130 @@
+    this.navigationBar = webview.getByLabel('PreferencesNavigation');
+
+    this.recipesCatalogButton = this.navigationBar.getByLabel('Recipes Catalog', { exact: true });
+    this.runningAppsButton = this.navigationBar.getByLabel('Running');
+    this.catalogButton = this.navigationBar.getByLabel('Catalog', { exact: true });
+    this.servicesButton = this.navigationBar.getByLabel('Services');
+    this.playgroundsButton = this.navigationBar.getByLabel('Playgrounds');
+
+    this.recipesCatalogPage = webview.getByRole('region', { name: 'Recipe Catalog' })
+    this.recipesCatalogContent = this.recipesCatalogPage.getByRole('region', { name: 'content', exact: true }).first();
+    this.recipesCatalogNaturalLanguageProcessing = this.recipesCatalogContent.getByRole('region', { name: 'Natural Language Processing', exact: true });
+    this.recipesCatalogAudio = this.recipesCatalogContent.getByRole('region', { name: 'Audio', exact: true });
+    this.recipesCatalogComputerVision = this.recipesCatalogContent.getByRole('region', { name: 'Computer Vision', exact: true });
+
+    this.startRecipeButton = this.webview.getByRole('button', { name: 'Start recipe' });
+    this.confirmStartRecipeButton = this.webview.getByRole('button', { name: /Start(\s+([A-Za-z]+\s+)+)recipe/i });
+    this.recipeStatus = this.webview.getByRole('status');
+
+    this.applicationDetailsPanel = webview.getByLabel('application details panel');
+    this.openAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Open AI App' });
+    this.deleteAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Delete AI App' });
+  }
+
+  async waitForPageLoad(): Promise<void> {
+    await playExpect(this.heading).toBeVisible();
+    await playExpect(this.navigationBar).toBeVisible();
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await playExpect(this.runningAppsButton).toBeVisible();
+    await playExpect(this.catalogButton).toBeVisible();
+    await playExpect(this.servicesButton).toBeVisible();
+    await playExpect(this.playgroundsButton).toBeVisible();
+  }
+
+  async openRecipesCatalog(): Promise<void> {
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await this.recipesCatalogButton.click();
+    await this.page.waitForTimeout(2000);
+    await playExpect(this.recipesCatalogPage).toBeVisible();
+    await playExpect(this.recipesCatalogContent).toBeVisible();
+    await playExpect(this.recipesCatalogNaturalLanguageProcessing).toBeVisible();
+    await playExpect(this.recipesCatalogAudio).toBeVisible();
+    await playExpect(this.recipesCatalogComputerVision).toBeVisible();
+  }
+
+  async openRecipesCatalogApp(category: Locator, appName: string): Promise<void> {
+    await this.openRecipesCatalog();
+    await category.getByRole('region', { name: 'content', exact: true }).first().getByRole('region', { name: appName, exact: true }).getByRole('button', { name: 'More details' }).click({ timeout: 5_000 });
```

Comment:
Default timeout for click() action is 5 seconds, there is no reason to provide it explicitly unless you change it to a different value.

Reply:
thank you for the suggestion @cbr7 updated in new page fragment